bundles/CoreBundle/Migrations/Version20211018104331.php

```diff
@@ -0,0 +1,110 @@
+<?php
+
+declare(strict_types=1);
+
+/**
+ * Pimcore
+ *
+ * This source file is available under two different licenses:
+ * - GNU General Public License version 3 (GPLv3)
+ * - Pimcore Commercial License (PCL)
+ * Full copyright and license information is available in
+ * LICENSE.md which is distributed with this source code.
+ *
+ *  @copyright  Copyright (c) Pimcore GmbH (http://www.pimcore.org)
+ *  @license    http://www.pimcore.org/license     GPLv3 and PCL
+ */
+
+namespace Pimcore\Bundle\CoreBundle\Migrations;
+
+use Doctrine\DBAL\Schema\Schema;
+use Doctrine\DBAL\Schema\SchemaException;
+use Doctrine\Migrations\AbstractMigration;
+use Pimcore\Model\Dao\AbstractDao;
+use Pimcore\Model\DataObject;
+
+final class Version20211018104331 extends AbstractMigration
+{
+    public function getDescription(): string
+    {
+        return '';
+    }
+
+    public function up(Schema $schema): void
+    {
+        $list = new DataObject\ClassDefinition\Listing();
+
+        foreach ($list as $class) {
+            $foreignKeys = $this->getForeignKeys($class);
+
+            foreach($foreignKeys as $table => $objectIdColumn) {
+                try {
+                    $tableSchema = $schema->getTable($table);
+                } catch(SchemaException $e) {
+                    continue;
+                }
+
+                $fkName = AbstractDao::getForeignKeyName($table, $objectIdColumn);
+                if (!$tableSchema->hasForeignKey($fkName)) {
+                    if($tableSchema->getColumn($objectIdColumn)->getCustomSchemaOption('unsigned') === false) {
```

Comment:
```suggestion
                    $column = $tableSchema->getColumn($objectIdColumn);
                    if(!$column->hasCustomSchemaOption('unsigned') || $column->getCustomSchemaOption('unsigned') === false) {
```
Otherwise this would result in `Warning: Undefined array key "unsigned" `